apps/web/src/server/applications/applications.router.js

```diff
@@ -7,12 +7,20 @@ import * as locals from './applications.locals.js';
 import applicationsCaseRouter from './case/applications-case.router.js';
 import applicationsCreateRouter from './create-new-case/applications-create.router.js';
 import applicationsSearchRouter from './search-results/applications-search.router.js';
+import { destroySessionS51 } from './case/s51/applications-s51.session.js';
 
 const router = createRouter();
 const domainRouter = createRouter({ mergeParams: true });
 
 router.use(filters.registerFilters);
 router.use(locals.registerLocals);
+router.use((request, _, next) => {
+	const { originalUrl, session } = request;
+	if (!originalUrl.includes('s51-advice/create')) {
+		destroySessionS51(session);
+	}
+	next();
+});
```

Comment:
PROS: this makes sure to erase the session whenever out of the s51 creation process

CONS: this bit of code is executed for _every_ page in the applications scope

Reply:
maybe check first if s51 data is in the session first - then do the rest if there is related data